File: Data Structures & Algorithms/anagram-groups/submission-2.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        def getFreq(s):
            freq = [0] * 26
            for ch in s:
                freq[ord(ch) - ord('a')] += 1
            return tuple(freq)
        
        freqmap = {}
        for word in strs:
            freq = getFreq(word)
            if freq in freqmap:
                freqmap[freq].append(word)
            else:
                freqmap[freq] = [word]
        
        return [freqmap[freq] for freq in freqmap]

File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        if(s.size() != t.size()) return false;

        int freqs[26] = {};
        for(char ch : s) freqs[(int(ch - 'a'))]++;
        
        for(char ch : t) freqs[(int(ch - 'a'))]--;

        for(int freq : freqs) {
            if(freq != 0) return false;
        }
        return true;
    }
};
